py/mpconfig.h

```diff
@@ -490,6 +490,11 @@
 #define MICROPY_LONGINT_IMPL (MICROPY_LONGINT_IMPL_NONE)
 #endif
 
+// Support for calls to buildin.pow() with 3 integer arguments
+#ifndef MICROPY_INT_POW3
+#define MICROPY_INT_POW3 (1)
```

Comment:
There's also MicroPython spec. It's no.1 is "MicroPython should be as small as possible". ;-)

> Also, the default behaviour before my patch had this functionality enabled;

And you initially had a check in the different place. You re-placed it correctly, and that helped to solve both your agenda of efficient algo always used for mpz's and my of disabling functionality used by <0.1% (I count by number of project stars by number of reports of this issue) vs decreasing the size for almost everyone ;-).

> but I really do think that the default should be to follow the Python spec.

(Un)fortunately, that's not possible. As a random (of hundreds) example, we aren't going to implement formally correct, also over-complex method resolution algorithm for multiple inheritance from big Python: https://www.python.org/download/releases/2.3/mro/ . And we have good foundation for that: there're language (some are still at no.1 at some ratings) which don't support multiple inheritance. We support some, and that's already cooler than those languages, even if not 100% like big Python.

> If maintainers seriously need to save space a much better option is to remove MPZ support.

Unfortunately, that's not how it works in the real world. It's easy to spot that pow3 is not available when you need and enable it. But majority of people would be lost when presented with a big monolith which doesn't fit their usecase. Here's a proof: I watch an experienced Linux maintainer to run following work for years: http://connect.linaro.org/resource/las16/las16-407/ . But apparently, he still didn't manage to make Linux tiny enough, so even "Linux Foundation" had to adopt another OS under its auspices for cases where Linux doesn't fit. Well, we learn from industrial lessons here ;-).

(All this commentary is for the change discussion, not argument purposes of course. There's a well-defined direction and paradigm for MicroPython, and interested parties can learn it (i.e. it's not random). As a usual disclaimer though, the above expresses my opinions in my rendition, and there may be differences with "official" project's opinions and directions.)